src/main/java/org/example/dao/RouteDAOImpl.java

```diff
@@ -0,0 +1,126 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+import org.example.model.Route;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RouteDAOImpl implements RouteDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public RouteDAOImpl() {
+    }
+
+    @Override
+    public void insertRoute(Route route) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try {
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO route (startPointId, endPointId, shortestDistance) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS);
+            statement.setInt(1, route.getStartPoint().getId());
+            statement.setInt(2, route.getEndPoint().getId());
+            statement.setLong(3, route.getShortestDistance());
+            statement.executeUpdate();
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                int id = generatedKeys.getInt(1);
+                route.setId(id);
+                System.out.println("Route inserted successfully. Generated ID: " + id);
+            }
+
+            statement.close();
+            generatedKeys.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public void updateRoute(Route route) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try {
+            PreparedStatement statement = connection.prepareStatement("UPDATE route SET shortestDistance = ? WHERE startPointId = ? AND endPointId = ?");
+            statement.setLong(1, route.getShortestDistance());
+            statement.setInt(2, route.getStartPoint().getId());
+            statement.setInt(3, route.getEndPoint().getId());
+            int rowsAffected = statement.executeUpdate();
+            if (rowsAffected > 0) {
+                System.out.println("Route updated successfully.");
+            } else {
+                System.out.println("Route not found.");
+            }
+            statement.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public Route getRoute(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        Route route = null;
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT * FROM route WHERE id = ?");
```

Comment:
I don't think this will change anything in this project in particular, but as I understood, we are expected to always use aliases instead of * even when it seems straight forward (at least, this was the last correction for my previous project), i.e., route r, r.id AS route_id, r.start_point AS route_start_point and so on

Reply:
done